litmus/AArch64Compile_litmus.ml

```diff
@@ -832,11 +834,44 @@ module Make(V:Constant.S)(C:Config) =
           reg_env = (add_128 [r1;r2;]) @ [(r3,voidstar)]}
     | _ -> assert false
 
-    let mov_simd_ve r1 i1 r2 i2 =
+    let addv_simd v r1 r2 =
+      { empty_ins with
+        memo = sprintf "addv %s, %s" (print_vecreg v "o" 0) (print_simd_reg "i" 0 0 r2);
+        inputs = [r2];
+        outputs = [r1];
+        reg_env = (add_128 [r1;r2])}
+
+    let dup_simd_v r1 v r2 =
+      { empty_ins with
+        memo = sprintf "dup %s, %s"
+          (print_simd_reg "o" 0 0 r1)
+          (match v with | V32 -> "^wi0" | V64 -> "^i0" | V128 -> assert false);
+        inputs = [r2];
+        outputs = [r1];
+        reg_env = (add_128 [r1;]) @ ((
+          match v with
+          | V32 -> add_w
+          | V64 -> add_q
+          | V128 -> assert false) [r2;])}
+
+    let fmov_simd_tg v1 r1 v2 r2 =
       { empty_ins with
-        memo = sprintf "mov %s[%i], %s[%i]" (print_simd_reg "o" 0 0 r1) i1 (print_simd_reg "i" 0 0 r2) i2;
+        memo = sprintf "fmov %s, %s"
+          (match v1 with | V32 -> "^wo0" | V64 -> "^o0" | V128 -> assert false)
+          (print_vecreg v2 "i" 0);
         inputs = [r2];
         outputs = [r1];
+        reg_env = ((match v1 with
+          | V32 -> add_w
+          | V64 -> add_q
+          | V128 -> assert false)
+        [r1;]) @ (add_128 [r2;])}
+
+    let mov_simd_ve r1 i1 r2 i2 =
+      { empty_ins with
+        memo = sprintf "mov %s[%i], %s[%i]" (print_simd_reg "o" 0 0 r1) i1 (print_simd_reg "i" 0 1 r2) i2;
+        inputs = [r1;r2]; (* r1 is intentionally in 'inputs', see comment for reg_class_stable *)
+        outputs = [];
```

Comment:
Something is wrong here:  the code snippet `(print_simd_reg "o" 0 0 r1)` refers to the first of output registers, while the list of output registers is empty (line 874).

Moreover I could not understand the comment.

Reply:
It would be safer to have `r1` in both lists, as, if I understand well `r1` is both read and written by the instruction.